### ISSUE TITLE
Wreath product regression

### ISSUE THREAD

**Issue by egri-nagy** [gap-packages/sgpdec#19]
With GAP 4.11 the wreath product construction is not working any more. It works fine with GAP 4.10.2.

```########> Diff in ~/gap-4.11.0/pkg/sgpdec/tst/wreath.tst:26
# Input is:
ForAll(triples, function(t)
 local A,B,C,AB,BC,ABC,AB_C,A_BC;
 A := t[1];
 B := t[2];
 C := t[3];
 AB := Image(IsomorphismPermGroup(GroupWreathProduct([A,B])));
 BC := Image(IsomorphismPermGroup(GroupWreathProduct([B,C])));
 ABC := Image(IsomorphismPermGroup(GroupWreathProduct([A,B,C])));
 AB_C := Image(IsomorphismPermGroup(GroupWreathProduct([AB,C])));
 A_BC := Image(IsomorphismPermGroup(GroupWreathProduct([A,BC])));
 return IsomorphismGroups(ABC,A_BC)<>fail
        and
        IsomorphismGroups(ABC,AB_C)<>fail; end);
# Expected output:
true
# But found:
false
########
```

**Comment by fingolfin** [comment]
To track this down might be helpful to do a `git bisect` on the GAP repository. If anyone wants to try, the script etc/bisect.sh included with GAP helps with that.

**Comment by egri-nagy** [comment]
In GAP 4.10.2:

```
gap> Z2:=CyclicGroup(IsPermGroup,2);
Group([ (1,2) ])
gap> Image(IsomorphismPermGroup(GroupWreathProduct([Z2,Z2])));
Group([ (1,3)(2,4), (1,2) ])
```
while in GAP 4.11 we get
```
ap> Z2:=CyclicGroup(IsPermGroup,2);
Group([ (1,2) ])
gap> Image(IsomorphismPermGroup(GroupWreathProduct([Z2,Z2])));
Group([ (1,5)(2,6)(3,7)(4,8), (1,3)(2,4)(5,6)(7,8) ])
```
which causes trouble since we use them as permutation groups.
I'll try whether asking for a  smaller representation fixes the test.

**Comment by egri-nagy** [comment]
fixed by 06eecb697f69ced7ab7c45b7df5ccf3f6c04780c

**Comment by nehaniv** [comment]
hi @,
Maybe there's a bug elsewhere - I just saw the discussion is
[semigroups/Semigroups]  from Tom Conti-Leslie and Wilf Wilson  , but maybe
it's not relevant, that it is sometimes giving wrong 'isomorphisms'  though
it seems to be a problem already with GAP 10.4.2.
Best,
X
